core/shared/src/main/scala/sttp/client/ResponseAs.scala

```diff
@@ -12,7 +12,7 @@ import scala.util.{Failure, Success, Try}
   * @tparam T Target type as which the response will be read.
   * @tparam S If `T` is a stream, the type of the stream. Otherwise, `Nothing`.
   */
-sealed trait ResponseAs[T, +S] {
+sealed trait ResponseAs[+T, +S] {
```

Comment:
Maybe you could add your original example e.g. to `RequestTests` just to know that the code compiles (the compilation is the test :) ). + a comment with the issue's number